pkg/kubelet/util/pod_startup_latency_tracker.go

```diff
@@ -150,8 +228,48 @@ func (p *basicPodStartupLatencyTracker) RecordImageFinishedPulling(podUID types.
 		return
 	}
 
-	if !state.firstStartedPulling.IsZero() {
-		state.lastFinishedPulling = p.clock.Now() // Now is always grater than values from the past.
+	now := p.clock.Now()
+
+	// Complete the oldest pull session if we have active starts
```

Comment:
why are we taking first oldest session here? Are we guaranteed to choose the correct image start time?